rocAL/include/pipeline/node.h

```diff
@@ -73,3 +75,62 @@ class Node {
         (this->_args.push_back(Argument(arg_names[Indices], std::forward<Args>(args))), ...);
     }
 };
+
+class NodeFactory {
+public:
+    using LoaderCreator = std::function<std::shared_ptr<Node>(Tensor*, void*)>;
```

Comment:
Blocking: `NodeFactory` now relies on `std::function`, but `rocAL/include/pipeline/node.h` does not include `<functional>`. Several translation units include `node.h` without pulling in `<functional>` transitively, so this header no longer compiles when built on its own. Please add the missing include.